### PR TITLE
[microNPU] Fix bug in channels extraction in the matcher

### PR DESCRIPTION
**Pull request by ekalda**
If the input tensor layout is in NHCWB16, we were passing W value
instead of the channels to get_valid_block_configs.

cc @Mousius @lhutton1

### CODE REVIEW

**Comment by ekalda** [comment]
cc: @manupa-arm @NicolaLancellotti

File: python/tvm/relay/backend/contrib/ethosu/te/pooling.py [hunk 1]
```diff
@@ -239,8 +239,8 @@ def match_ethosu_pooling(output_tensor, device_config):
     ifm_dtype = input_tensors[0].dtype
     ofm_dtype = output_tensor.dtype
 
-    ifm_channels = int(input_tensors[0].shape[3])
-    ofm_channels = ifm_channels
+    # Use channels form a stage of TE graph where the IFM is always NHWC
```
Comment (lhutton1) [inline python/tvm/relay/backend/contrib/ethosu/te/pooling.py:242]: Nit: s/form/from

**Review by lhutton1 — COMMENTED** [review]
Thanks for the fix @ekalda! Is it possible to add a test to exercise this case?

**Review by lhutton1 — APPROVED** [review]
LGTM, thanks for digging into it!

**Comment by manupa-arm** [comment]
Thanks @ekalda @lhutton1 !

**Pull request merged**